pennylane/templates/subroutines/flip_sign.py

```diff
@@ -0,0 +1,143 @@
+        bin_arr (array[int]): binary array vector representing the state to flip the sign
+
+    Raises:
+        ValueError: "expected at integer binary array "
+        ValueError: "expected at integer binary array for wires "
+        ValueError: "expected at integer binary array not empty "
+        ValueError: "expected at least one wire representing the qubit "
+
+    .. seealso:: :func:`~.relevant_func`, :class:`~.RelevantClass` (optional)
+
+    .. details::
+
+        :title: Usage Details
+
+        The template is used inside a qnode.
+        The number of shots has to be explicitly set on the device when using sample-based measurements:
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit", wires=5, shots = 1000)
+
+            @qml.qnode(dev)
+            def circuit():
+               for wire in list(range(5)):
+                    qml.Hadamard(wires = wire)
+               qml.FlipSign([1,0,1,0,0], wires = list(range(5)))
+               for wire in list(range(5)):
+                    qml.Hadamard(wires = wire)
+               return qml.sample()
+
+            drawer = qml.draw(circuit, show_all_wires = True)
+
+        The result for the above circuit is:
+
+            >>> print(drawer())
+            0: ──H─╭FlipSign──H─┤  Sample
+            1: ──H─├FlipSign──H─┤  Sample
+            2: ──H─├FlipSign──H─┤  Sample
+            3: ──H─├FlipSign──H─┤  Sample
+            4: ──H─╰FlipSign──H─┤  Sample
+
+    """
+
+    num_wires = AnyWires
+
+    def __init__(self, bin_arr, wires, do_queue=True, id=None):
+
+        if not isinstance(bin_arr, list):
+            raise ValueError("expected at integer binary array ")
+
+        if np.array(bin_arr).dtype != np.dtype("int"):
+            raise ValueError("expected at integer binary array ")
+
+        if not isinstance(wires, list):
+            raise ValueError("expected at integer binary array for wires ")
+
+        if np.array(wires).dtype != np.dtype("int"):
+            raise ValueError("expected a integer binary array for wires ")
+
+        if len(bin_arr) == 0:
+            raise ValueError("expected at integer binary array not empty ")
+
+        if len(wires) == 0:
+            raise ValueError("expected at least one wire representing the qubit ")
+
+        self._hyperparameters = {"bin_arr": bin_arr}
+        super().__init__(wires=wires, do_queue=do_queue, id=id)
+
+    @property
+    def num_params(self):
+        return 0
+
+    @staticmethod
+    def compute_decomposition(wires, bin_arr):
+        r"""Representation of the operator
+
+        .. math::
+            \hat{X} \otimes \hat{Z} \otimes \hat{X}
+
+        .. seealso:: :meth:`~.FlipSign.decomposition`.
+
+        Args:
+            wires (array[int]): wires that the operator acts on
+            bin_arr (array[int]): binary array vector representing the state to flip the sign
+
+        Returns:
+            list[Operator]: decomposition of the operator
+        """
+
```

Comment:
this is where you should ask if `n` is integer or list. If it is an integer, with the script that I left you in the issue you pass it to list 💪

Reply:
I have implemented this with your code suggestion, let me hear what you think about...👍